src/codegen.cpp

```diff
@@ -7607,7 +7606,7 @@ jl_compile_result_t jl_emit_code(
         compare_cgparams(params.params, &jl_default_cgparams)) &&
         "functions compiled with custom codegen params must not be cached");
     JL_TRY {
-        std::tie(m, decls) = emit_function(li, src, jlrettype, params, jl_LLVMContext);
+        std::tie(m, decls) = emit_function(li, src, jlrettype, params, *jl_ExecutionEngine->getContext().getContext());
```

Comment:
this should perhaps come from the jl_codegen_params_t argument, since almost everything in there is derived from/associated with an LLVMContext

Reply:
otherwise, this seems a bit like cheating, since the end goal is to prevent jl_ExecutionEngine from having a context

Reply:
I've now lifted those jl_ExecutionEngine calls all the way to the closest JL_LOCK(&codegen_lock) statements, but do we want to replace them by creating a new LLVMContext every time we codegen?